File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        stack = []
        for i, val in enumerate(temperatures):
            while stack and stack[-1][0] < val:
                sV , sI = stack.pop()
                res[sI] = (i - sI)
            stack.append([val, i])
        return res

File: Data Structures & Algorithms/find-k-closest-elements/submission-0.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
         priority_queue<int, vector<int>, greater<int>> heap;
         for (int num: arr) {
            if (heap.size() < k){
                heap.push(num);
            } else if (abs(num-x) < abs(heap.top() - x) || 
            (abs(num-x) == abs(heap.top() - x) && (num < x))) {
                heap.pop();
                heap.push(num);
            }
         }
        vector<int> res;
        while (!heap.empty()) { 
            res.push_back(heap.top());
            heap.pop();
        }
        sort(res.begin(), res.end());
        return res;
    }
};